### PR TITLE
Fix ParseTolerant when zeros are continuous

### LINKED ISSUE #55: ParseTolerant now fails on a valid semver
After #47  I noticed that the following (which semver.ParseTolerant had previously accepted) now no longer parses:

semver.Parse("4.1.0-0.nightly") is returned as 4.1.0-0.nightly

but after this change

semver.ParseTolerant("4.1.0-0.nightly") errors with strconv.ParseUint: parsing "": invalid syntax

I think the fix is that the "remove leading zeroes" part of that change needs a more accurate check than `len(p) >1` - probably needs to be looking to the first dot, not the whole string.

### PR DESCRIPTION
**Pull request by Songmu**
It fixes #55. This is a serious bug, so I want you to respond early.

### CODE REVIEW

**Comment by Songmu** [comment]
@blang How about it?

**Comment by lootek** [comment]
upvote merging, got hit by the same issue...

File: semver.go [hunk 1]
```diff
@@ -243,7 +243,11 @@ func ParseTolerant(s string) (Version, error) {
 	// Remove leading zeros.
 	for i, p := range parts {
 		if len(p) > 1 {
-			parts[i] = strings.TrimPrefix(p, "0")
+			p = strings.TrimLeft(p, "0")
+			if len(p) == 0 || !strings.ContainsAny(p[0:1], "0123456789") {
```
Comment (ferhatelmas) [inline semver.go]: might compare to empty string and drop `0` such that

`if p == "" || !strings.ContainsAny(p[:1], "123456789) {`

**Comment by blang** [comment]
Thanks for your contribution and long wait!

**Pull request merged**